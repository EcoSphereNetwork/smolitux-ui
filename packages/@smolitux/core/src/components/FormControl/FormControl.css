.smolitux-form-control {
  display: flex;
  flex-direction: column;
  width: 100%;
  margin-bottom: 1rem;
}

/* Label-Positionen */
.smolitux-form-control--label-top {
  flex-direction: column;
}

.smolitux-form-control--label-left {
  flex-direction: row;
  align-items: center;
}

.smolitux-form-control--label-left .smolitux-form-label {
  margin-right: 1rem;
  margin-bottom: 0;
  min-width: 8rem;
  text-align: right;
}

.smolitux-form-control--label-right {
  flex-direction: row-reverse;
  align-items: center;
}

.smolitux-form-control--label-right .smolitux-form-label {
  margin-left: 1rem;
  margin-bottom: 0;
  min-width: 8rem;
}

.smolitux-form-control--label-bottom {
  flex-direction: column-reverse;
}

.smolitux-form-control--label-bottom .smolitux-form-label {
  margin-top: 0.5rem;
  margin-bottom: 0;
}

.smolitux-form-control--label-floating {
  position: relative;
}

.smolitux-form-control--label-floating .smolitux-form-label {
  position: absolute;
  top: 0;
  left: 0;
  padding: 0.5rem 0.75rem;
  pointer-events: none;
  transition: all 0.2s ease-in-out;
  transform-origin: left top;
  background-color: transparent;
  z-index: 1;
}

.smolitux-form-control--label-floating.smolitux-form-control--focused .smolitux-form-label,
.smolitux-form-control--label-floating.smolitux-form-control--filled .smolitux-form-label {
  transform: translateY(-1.25rem) scale(0.85);
  color: var(--primary-600, #2563eb);
}

/* Form Label */
.smolitux-form-label {
  font-size: 0.875rem;
  font-weight: 500;
  color: var(--form-label-color, #374151);
  margin-bottom: 0.5rem;
  display: flex;
  align-items: center;
}

.smolitux-form-label--disabled {
  color: var(--form-label-disabled-color, #9ca3af);
  cursor: not-allowed;
}

.smolitux-form-label__required-indicator {
  color: var(--form-required-color, #ef4444);
  margin-left: 0.25rem;
}

.smolitux-form-label__optional-indicator {
  color: var(--form-optional-color, #6b7280);
  font-size: 0.75rem;
  font-weight: normal;
  margin-left: 0.25rem;
}

/* Helper Text */
.smolitux-form-helper-text {
  font-size: 0.75rem;
  color: var(--form-helper-text-color, #6b7280);
  margin-top: 0.5rem;
}

.smolitux-form-helper-text--disabled {
  color: var(--form-helper-text-disabled-color, #9ca3af);
}

/* Error Message */
.smolitux-form-error-message {
  font-size: 0.75rem;
  color: var(--form-error-color, #ef4444);
  margin-top: 0.5rem;
  display: flex;
  align-items: center;
}

.smolitux-form-error-message__icon {
  margin-right: 0.25rem;
}

/* Success Message */
.smolitux-form-success-message {
  font-size: 0.75rem;
  color: var(--form-success-color, #10b981);
  margin-top: 0.5rem;
  display: flex;
  align-items: center;
}

.smolitux-form-success-message__icon {
  margin-right: 0.25rem;
}

/* Größen */
.smolitux-form-control--xs .smolitux-form-label {
  font-size: 0.75rem;
  margin-bottom: 0.25rem;
}

.smolitux-form-control--sm .smolitux-form-label {
  font-size: 0.8125rem;
  margin-bottom: 0.375rem;
}

.smolitux-form-control--lg .smolitux-form-label {
  font-size: 1rem;
  margin-bottom: 0.625rem;
}

/* Zustände */
.smolitux-form-control--disabled {
  opacity: 0.7;
  cursor: not-allowed;
}

.smolitux-form-control--invalid .smolitux-form-label {
  color: var(--form-error-color, #ef4444);
}

.smolitux-form-control--valid .smolitux-form-label {
  color: var(--form-success-color, #10b981);
}

.smolitux-form-control--success .smolitux-form-label {
  color: var(--form-success-color, #10b981);
}

.smolitux-form-control--loading {
  position: relative;
}

.smolitux-form-control--loading::after {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(255, 255, 255, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 1;
}

.smolitux-form-control--loading::before {
  content: '';
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 1.5rem;
  height: 1.5rem;
  border: 2px solid var(--primary-200, #bfdbfe);
  border-top-color: var(--primary-500, #3b82f6);
  border-radius: 50%;
  animation: smolitux-form-control-spin 1s linear infinite;
  z-index: 2;
}

@keyframes smolitux-form-control-spin {
  to {
    transform: translate(-50%, -50%) rotate(360deg);
  }
}

/* Varianten */
.smolitux-form-control--filled .smolitux-form-input {
  background-color: var(--form-filled-bg, #f3f4f6);
  border-color: transparent;
}

.smolitux-form-control--filled .smolitux-form-input:focus {
  background-color: var(--form-filled-focus-bg, #e5e7eb);
}

.smolitux-form-control--outlined .smolitux-form-input {
  background-color: transparent;
  border-width: 2px;
}

.smolitux-form-control--unstyled .smolitux-form-input {
  background-color: transparent;
  border: none;
  padding: 0;
}

/* Dark Mode */
.dark .smolitux-form-label {
  color: var(--form-label-color-dark, #e5e7eb);
}

.dark .smolitux-form-label--disabled {
  color: var(--form-label-disabled-color-dark, #6b7280);
}

.dark .smolitux-form-label__optional-indicator {
  color: var(--form-optional-color-dark, #9ca3af);
}

.dark .smolitux-form-helper-text {
  color: var(--form-helper-text-color-dark, #9ca3af);
}

.dark .smolitux-form-helper-text--disabled {
  color: var(--form-helper-text-disabled-color-dark, #6b7280);
}

.dark .smolitux-form-control--loading::after {
  background-color: rgba(31, 41, 55, 0.5);
}

.dark .smolitux-form-control--filled .smolitux-form-input {
  background-color: var(--form-filled-bg-dark, #374151);
}

.dark .smolitux-form-control--filled .smolitux-form-input:focus {
  background-color: var(--form-filled-focus-bg-dark, #4b5563);
}