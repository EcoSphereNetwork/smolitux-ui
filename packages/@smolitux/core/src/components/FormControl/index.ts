export { FormControl, useFormControl } from './FormControl';
export type { FormControlProps } from './FormControl';

// Für Abwärtskompatibilität mit dem bestehenden Export
export { FormControl as default } from './FormControl';