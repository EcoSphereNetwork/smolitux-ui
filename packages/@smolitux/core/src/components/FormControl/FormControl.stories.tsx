import type { Meta, StoryObj } from '@storybook/react';
import { FormControl } from './FormControl';

const meta: Meta<typeof FormControl> = {
  title: 'Components/FormControl',
  component: FormControl,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: 'FormControl',
  },
};

export const CustomStyle: Story = {
  args: {
    children: 'Custom FormControl',
    className: 'custom-style',
  },
};

export const Interactive: Story = {
  args: {
    children: 'Interactive FormControl',
    onClick: () => alert('Clicked!'),
  },
};
