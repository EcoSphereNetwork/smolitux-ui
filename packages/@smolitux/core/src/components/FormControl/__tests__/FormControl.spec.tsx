import React from 'react';
import renderer from 'react-test-renderer';
import { FormControl } from '../FormControl';

describe('FormControl Snapshots', () => {
  test('renders correctly with default props', () => {
    const tree = renderer.create(
      <FormControl>
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with label', () => {
    const tree = renderer.create(
      <FormControl label="Test Label">
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with helper text', () => {
    const tree = renderer.create(
      <FormControl helperText="Helper Text">
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with error message', () => {
    const tree = renderer.create(
      <FormControl error="Error Message">
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with success message', () => {
    const tree = renderer.create(
      <FormControl successMessage="Success Message">
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with required indicator', () => {
    const tree = renderer.create(
      <FormControl label="Test Label" required>
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with label position top', () => {
    const tree = renderer.create(
      <FormControl label="Test Label" labelPosition="top">
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with label position left', () => {
    const tree = renderer.create(
      <FormControl label="Test Label" labelPosition="left">
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with label position right', () => {
    const tree = renderer.create(
      <FormControl label="Test Label" labelPosition="right">
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with label position bottom', () => {
    const tree = renderer.create(
      <FormControl label="Test Label" labelPosition="bottom">
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with label position floating', () => {
    const tree = renderer.create(
      <FormControl label="Test Label" labelPosition="floating">
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with full width', () => {
    const tree = renderer.create(
      <FormControl fullWidth>
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with counter', () => {
    const tree = renderer.create(
      <FormControl showCounter counterValue={5} counterMax={10}>
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with progress bar', () => {
    const tree = renderer.create(
      <FormControl showProgressBar progressValue={50} progressMax={100}>
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with loading indicator', () => {
    const tree = renderer.create(
      <FormControl isLoading showLoadingIndicator>
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with error indicator', () => {
    const tree = renderer.create(
      <FormControl error="Error" showErrorIndicator>
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with success indicator', () => {
    const tree = renderer.create(
      <FormControl isSuccess showSuccessIndicator>
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with validation indicator', () => {
    const tree = renderer.create(
      <FormControl isValid showValidationIndicator>
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with hidden label', () => {
    const tree = renderer.create(
      <FormControl label="Hidden Label" hideLabel>
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with description for screen readers', () => {
    const tree = renderer.create(
      <FormControl description="Description for screen readers">
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with custom class names', () => {
    const tree = renderer.create(
      <FormControl 
        className="custom-class"
        containerClassName="container-class"
        labelClassName="label-class"
        helperTextClassName="helper-class"
        errorClassName="error-class"
        successClassName="success-class"
        fieldContainerClassName="field-container-class"
        label="Label"
        helperText="Helper"
      >
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with tooltip', () => {
    const tree = renderer.create(
      <FormControl 
        tooltip="Tooltip Text"
        labelTooltip="Label Tooltip"
      >
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with multiple props combined', () => {
    const tree = renderer.create(
      <FormControl 
        label="Combined Label"
        helperText="Helper Text"
        required
        labelPosition="floating"
        fullWidth
        size="lg"
        variant="filled"
        showCounter
        counterValue={5}
        counterMax={10}
        showProgressBar
        progressValue={50}
        progressMax={100}
        isLoading
        showLoadingIndicator
        tooltip="Tooltip Text"
      >
        <input />
      </FormControl>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});