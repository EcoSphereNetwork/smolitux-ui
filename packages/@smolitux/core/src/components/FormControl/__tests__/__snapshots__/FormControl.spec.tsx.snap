// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormControl Snapshots renders correctly with counter 1`] = `
<div
  className="
          
          
          
          
        "
>
  <div
    className=""
  >
    <div
      className="relative "
    >
      <input />
    </div>
    <div
      className="mt-1 text-xs text-right"
    >
      <span
        className="text-gray-500"
      >
        5
        /10
      </span>
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with custom class names 1`] = `
<div
  className="
          
          
          custom-class
          container-class
        "
>
  <label
    className="text-sm font-medium text-gray-700 dark:text-gray-300 block label-class mb-1"
    htmlFor="form-control-fo4789t"
    style={{}}
  >
    Label
  </label>
  <div
    className="field-container-class"
  >
    <div
      className="relative "
    >
      <input />
    </div>
    <div
      className="mt-1 text-sm"
    >
      <p
        className="text-gray-500 dark:text-gray-400 helper-class"
        id="form-control-fo4789t-helper"
      >
        Helper
      </p>
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with default props 1`] = `
<div
  className="
          
          
          
          
        "
>
  <div
    className=""
  >
    <div
      className="relative "
    >
      <input />
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with description for screen readers 1`] = `
<div
  className="
          
          
          
          
        "
>
  <div
    className=""
  >
    <div
      className="sr-only"
      id="form-control-caalisf-description"
    >
      Description for screen readers
    </div>
    <div
      className="relative "
    >
      <input />
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with error indicator 1`] = `
<div
  className="
          
          
          
          
        "
>
  <div
    className=""
  >
    <div
      className="relative "
    >
      <input />
      <div
        className="absolute right-3 top-1/2 transform -translate-y-1/2"
      >
        <span
          aria-hidden="true"
          className="text-red-500"
        >
          ✕
        </span>
      </div>
    </div>
    <div
      className="mt-1 text-sm"
    >
      <p
        className="text-red-600 dark:text-red-400 "
        id="form-control-2jfukjc-error"
        role="alert"
      >
        Error
      </p>
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with error message 1`] = `
<div
  className="
          
          
          
          
        "
>
  <div
    className=""
  >
    <div
      className="relative "
    >
      <input />
      <div
        className="absolute right-3 top-1/2 transform -translate-y-1/2"
      >
        <span
          aria-hidden="true"
          className="text-red-500"
        >
          ✕
        </span>
      </div>
    </div>
    <div
      className="mt-1 text-sm"
    >
      <p
        className="text-red-600 dark:text-red-400 "
        id="form-control-raqtbor-error"
        role="alert"
      >
        Error Message
      </p>
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with full width 1`] = `
<div
  className="
          w-full
          
          
          
        "
>
  <div
    className=""
  >
    <div
      className="relative "
    >
      <input />
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with helper text 1`] = `
<div
  className="
          
          
          
          
        "
>
  <div
    className=""
  >
    <div
      className="relative "
    >
      <input />
    </div>
    <div
      className="mt-1 text-sm"
    >
      <p
        className="text-gray-500 dark:text-gray-400 "
        id="form-control-kz4x4h1-helper"
      >
        Helper Text
      </p>
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with hidden label 1`] = `
<div
  className="
          
          
          
          
        "
>
  <label
    className="text-sm font-medium text-gray-700 dark:text-gray-300 sr-only  mb-1"
    htmlFor="form-control-04n8w87"
    style={{}}
  >
    Hidden Label
  </label>
  <div
    className=""
  >
    <div
      className="relative "
    >
      <input />
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with label 1`] = `
<div
  className="
          
          
          
          
        "
>
  <label
    className="text-sm font-medium text-gray-700 dark:text-gray-300 block  mb-1"
    htmlFor="form-control-94i6j3p"
    style={{}}
  >
    Test Label
  </label>
  <div
    className=""
  >
    <div
      className="relative "
    >
      <input />
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with label position bottom 1`] = `
<div
  className="
          
          flex flex-col-reverse
          
          
        "
>
  <label
    className="text-sm font-medium text-gray-700 dark:text-gray-300 block  mt-1"
    htmlFor="form-control-hoojyc6"
    style={{}}
  >
    Test Label
  </label>
  <div
    className=""
  >
    <div
      className="relative "
    >
      <input />
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with label position floating 1`] = `
<div
  className="
          
          relative
          
          
        "
>
  <label
    className="text-sm font-medium text-gray-700 dark:text-gray-300 block  absolute left-2 top-0 px-1 bg-white dark:bg-gray-800 text-xs transform -translate-y-1/2 pointer-events-none transition-all duration-200"
    htmlFor="form-control-je2ezh2"
    style={{}}
  >
    Test Label
  </label>
  <div
    className=""
  >
    <div
      className="relative pt-2"
    >
      <input />
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with label position left 1`] = `
<div
  className="
          
          flex items-start
          
          
        "
>
  <label
    className="text-sm font-medium text-gray-700 dark:text-gray-300 block  pt-2 pr-4"
    htmlFor="form-control-yblhk0y"
    style={
      {
        "width": "33%",
      }
    }
  >
    Test Label
  </label>
  <div
    className="flex-1"
  >
    <div
      className="relative "
    >
      <input />
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with label position right 1`] = `
<div
  className="
          
          flex flex-row-reverse items-start
          
          
        "
>
  <label
    className="text-sm font-medium text-gray-700 dark:text-gray-300 block  pt-2 pl-4"
    htmlFor="form-control-yhieyya"
    style={
      {
        "width": "33%",
      }
    }
  >
    Test Label
  </label>
  <div
    className="flex-1"
  >
    <div
      className="relative "
    >
      <input />
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with label position top 1`] = `
<div
  className="
          
          
          
          
        "
>
  <label
    className="text-sm font-medium text-gray-700 dark:text-gray-300 block  mb-1"
    htmlFor="form-control-35ru2d1"
    style={{}}
  >
    Test Label
  </label>
  <div
    className=""
  >
    <div
      className="relative "
    >
      <input />
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with loading indicator 1`] = `
<div
  className="
          
          
          
          
        "
>
  <div
    className=""
  >
    <div
      className="relative "
    >
      <input />
      <div
        className="absolute right-3 top-1/2 transform -translate-y-1/2"
      >
        <span
          aria-hidden="true"
          className="text-primary-500 animate-spin"
        >
          ⟳
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with multiple props combined 1`] = `
<div
  className="
          w-full
          relative
          
          
        "
  title="Tooltip Text"
>
  <label
    className="text-sm font-medium text-gray-700 dark:text-gray-300 block  absolute left-2 top-0 px-1 bg-white dark:bg-gray-800 text-xs transform -translate-y-1/2 pointer-events-none transition-all duration-200"
    htmlFor="form-control-4i10coc"
    style={{}}
  >
    Combined Label
    <span
      aria-hidden="true"
      className="ml-1 text-red-500"
    >
      *
    </span>
  </label>
  <div
    className=""
  >
    <div
      className="relative pt-2"
    >
      <input />
      <div
        className="absolute right-3 top-1/2 transform -translate-y-1/2"
      >
        <span
          aria-hidden="true"
          className="text-primary-500 animate-spin"
        >
          ⟳
        </span>
      </div>
    </div>
    <div
      className="mt-1 text-xs text-right"
    >
      <span
        className="text-gray-500"
      >
        5
        /10
      </span>
    </div>
    <div
      className="mt-2 h-1 w-full bg-gray-200 dark:bg-gray-700 rounded-full overflow-hidden"
    >
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={50}
        className="h-full bg-primary-500 transition-all duration-300 ease-in-out"
        role="progressbar"
        style={
          {
            "width": "50%",
          }
        }
      />
    </div>
    <div
      className="mt-1 text-sm"
    >
      <p
        className="text-gray-500 dark:text-gray-400 "
        id="form-control-4i10coc-helper"
      >
        Helper Text
      </p>
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with progress bar 1`] = `
<div
  className="
          
          
          
          
        "
>
  <div
    className=""
  >
    <div
      className="relative "
    >
      <input />
    </div>
    <div
      className="mt-2 h-1 w-full bg-gray-200 dark:bg-gray-700 rounded-full overflow-hidden"
    >
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={50}
        className="h-full bg-primary-500 transition-all duration-300 ease-in-out"
        role="progressbar"
        style={
          {
            "width": "50%",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with required indicator 1`] = `
<div
  className="
          
          
          
          
        "
>
  <label
    className="text-sm font-medium text-gray-700 dark:text-gray-300 block  mb-1"
    htmlFor="form-control-v4t7xw3"
    style={{}}
  >
    Test Label
    <span
      aria-hidden="true"
      className="ml-1 text-red-500"
    >
      *
    </span>
  </label>
  <div
    className=""
  >
    <div
      className="relative "
    >
      <input />
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with success indicator 1`] = `
<div
  className="
          
          
          
          
        "
>
  <div
    className=""
  >
    <div
      className="relative "
    >
      <input />
      <div
        className="absolute right-3 top-1/2 transform -translate-y-1/2"
      >
        <span
          aria-hidden="true"
          className="text-green-500"
        >
          ✓
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with success message 1`] = `
<div
  className="
          
          
          
          
        "
>
  <div
    className=""
  >
    <div
      className="relative "
    >
      <input />
    </div>
    <div
      className="mt-1 text-sm"
    >
      <p
        className="text-green-600 dark:text-green-400 "
        id="form-control-1495pci-success"
      >
        Success Message
      </p>
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with tooltip 1`] = `
<div
  className="
          
          
          
          
        "
  title="Tooltip Text"
>
  <div
    className=""
  >
    <div
      className="relative "
    >
      <input />
    </div>
  </div>
</div>
`;

exports[`FormControl Snapshots renders correctly with validation indicator 1`] = `
<div
  className="
          
          
          
          
        "
>
  <div
    className=""
  >
    <div
      className="relative "
    >
      <input />
      <div
        className="absolute right-3 top-1/2 transform -translate-y-1/2"
      >
        <span
          aria-hidden="true"
          className="text-green-500"
        >
          ✓
        </span>
      </div>
    </div>
  </div>
</div>
`;
